### ISSUE TITLE
Koin + Ktor + Live Reload With Gradle Throws `KoinApplication has not been started`

### ISSUE THREAD

**Issue by agrosner** [InsertKoinIO/koin#1045]
**Describe the bug**
Following the instructions to set up [Koin with Ktor](https://insert-koin.io/docs/reference/koin-ktor/ktor/)

I also have Ktor live reloading via `./gradlew built -t`

```kotlin

fun Application.module() {
    koin {
        slf4jLogger()
        modules(appModule)
    }
    val authController by inject<AuthController>()
     install(Authentication) {
        basic(BasicConfiguration) {
            realm = "**realm***"
            validate { credentials ->
                // fails when i access it here after live reload
                authController.validateCredentials(credentials)
            }
        }
    }
}
```

**To Reproduce**
1. Run both the live reloading build command, and start ktor app
2. Trying to authenticate via basic auth, accessing `authController` throws the error `KoinApplication has not been started`. *Note* this works if I manually restart the app.

**Expected behavior**
Should work on live reloads. 

When I access the `GlobalContext` in the debugger, the `KoinApplication` is there, but the `Koin` instance is not. It seems that somewhere along the way, `Koin` is not getting reinitialized.

**Koin project used and used version (please complete the following information):**
Koin Version 3.0.1-beta-1
Ktor 1.5.2

**Comment by agrosner** [comment]
by modifying that line as follows:
```kotlin
 val koinApplication by lazy {
                // PATCHES MISSING KOIN
                if (GlobalContext.getOrNull() == null) {
                    return@lazy startKoin(appDeclaration = configure)
                }
                GlobalContext.getKoinApplicationOrNull() ?: startKoin(appDeclaration = configure)
            }
```
I patched the bug. It doesn't have to be a `lazy` field, but I just used its convenience for initialization

**Comment by arnaudgiuliani** [comment]
And just that?

```kotlin
val koinApplication by lazy {
              GlobalContext.getKoinApplicationOrNull() ?: startKoin(appDeclaration = configure)
          }
```

**Comment by arnaudgiuliani** [comment]
can you test in beta-2?

**Comment by arnaudgiuliani** [comment]
SHould be fixed in 3.0.1
